File: src/pattern/prototype/java/UnderlinePen.java
package pattern.prototype.java;

public class UnderlinePen implements Product{
    private char ulchar;
    public UnderlinePen(char ulchar){
        this.ulchar = ulchar;
    }

    public void use(String s){
        int length = s.length();
        System.out.println(s);
        for (int i = 0; i < length; i++) {
            System.out.print(this.ulchar);
        }
        System.out.println();
    }

    public Product createClone(){
        Product p = null;

        try {
            p = (Product)clone();
        } catch (CloneNotSupportedException e) {
            e.printStackTrace();
        }

        return p;
    }
}

File: src/pattern/prototype/kotlin/UnderlinePen.kt
package pattern.prototype.kotlin

class UnderlinePen(val ulchar: Char): Product{

    override fun use(s: String) {
        val length = s.length
        println(s)
        for(i in 1..length){
            print(this.ulchar)
        }

        println()
    }

    override fun createClone(): Product? {
        var p: Product? = null
        try{
            p = this.clone() as Product
        } catch (e: CloneNotSupportedException){
            e.printStackTrace()
        }

        return p
    }
}